filters/include/pcl/filters/impl/filter_indices.hpp

```diff
@@ -74,7 +74,34 @@ pcl::removeNaNFromPointCloud (const pcl::PointCloud<PointT> &cloud_in,
   }
 }
 
+template<typename PointT> void
+pcl::FilterIndicesXYZ<PointT>::applyFilter (PointCloud &output)
+{
+  std::vector<int> indices;
+  if (keep_organized_)
+  {
+    bool temp = extract_removed_indices_;
+    extract_removed_indices_ = true;
+    applyFilter (indices);
+    extract_removed_indices_ = temp;
+
+    output = *input_;
+    for (const auto ri : *removed_indices_)  // ri = removed index
+      output[ri].getArray3fMap () = user_filter_value_;
+    if (!std::isfinite (user_filter_value_))
+      output.is_dense = false;
+  }
+  else
+  {
+    output.is_dense = true;
+    applyFilter (indices);
+    pcl::copyPointCloud (*input_, indices, output);
+  }
+}
+
+
 #define PCL_INSTANTIATE_removeNanFromPointCloud(T) template PCL_EXPORTS void pcl::removeNaNFromPointCloud<T>(const pcl::PointCloud<T>&, std::vector<int>&);
+#define PCL_INSTANTIATE_FilterIndicesXYZ(T) template class PCL_EXPORTS  pcl::FilterIndicesXYZ<T>;
```

Comment:
Why are we instantiating an incomplete class? `applyFilter (std::vector<int> &indices)` is pure virtual even in `FilterIndicesXYZ`

Reply:
I thought the same... until this happened
```
[100%] Linking CXX executable ../bin/pcl_passthrough_filter                                                                                                                           
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointXYZI>::applyFilter(pcl::PointCloud<pcl::PointXYZI>&)'                                     
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointXYZRGB>::applyFilter(pcl::PointCloud<pcl::PointXYZRGB>&)'                                 
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointXYZRGBA>::applyFilter(pcl::PointCloud<pcl::PointXYZRGBA>&)'                               
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointWithScale>::applyFilter(pcl::PointCloud<pcl::PointWithScale>&)'                           
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointXYZLNormal>::applyFilter(pcl::PointCloud<pcl::PointXYZLNormal>&)'                         
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointWithRange>::applyFilter(pcl::PointCloud<pcl::PointWithRange>&)'                           
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointXYZRGBNormal>::applyFilter(pcl::PointCloud<pcl::PointXYZRGBNormal>&)'                     
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointXYZHSV>::applyFilter(pcl::PointCloud<pcl::PointXYZHSV>&)'                                 
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointXYZINormal>::applyFilter(pcl::PointCloud<pcl::PointXYZINormal>&)'                         
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointWithViewpoint>::applyFilter(pcl::PointCloud<pcl::PointWithViewpoint>&)'                   
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::InterestPoint>::applyFilter(pcl::PointCloud<pcl::InterestPoint>&)'                             
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointXYZ>::applyFilter(pcl::PointCloud<pcl::PointXYZ>&)'                                       
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointSurfel>::applyFilter(pcl::PointCloud<pcl::PointSurfel>&)'                                 
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointXYZL>::applyFilter(pcl::PointCloud<pcl::PointXYZL>&)'                                     
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointDEM>::applyFilter(pcl::PointCloud<pcl::PointDEM>&)'                                       
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointNormal>::applyFilter(pcl::PointCloud<pcl::PointNormal>&)'                                 
../lib/libpcl_filters.so.1.10.1.99: undefined reference to `pcl::FilterIndicesXYZ<pcl::PointXYZRGBL>::applyFilter(pcl::PointCloud<pcl::PointXYZRGBL>&)'
```

The implementation of this function still needs to be compiled, despite the user not being able to construct a `FilterIndicesXYZ` object.

Reply:
Oh wow